rig/machine.py

```diff
@@ -0,0 +1,220 @@
+"""Definitions of the resources available in a SpiNNaker machine.
+
+Overview
+--------
+
+* :py:class:`~.rig.machine.Links` gives identifiers for links in a machine
+
+* :py:class:`~.rig.machine.Cores`, :py:class:`~.rig.machine.SDRAM` and
+  :py:class:`~.rig.machine.SDRAM` are (suggested) identifiers for common
+  resources in a SpiNNaker machine.
+
+* :py:class:`~.rig.machine.Machine` defines resources what resources are
+  available in a machine.
+
+"""
+
+from enum import Enum, IntEnum
+
+import sentinel
+
+
+class Links(IntEnum):
+    """Enumeration of links from a SpiNNaker chip.
+
+    Note that the numbers chosen have two useful properties:
+    * The integer values assigned are chosen to match the numbers used to
+      identify the links in the low-level software API and hardware registers.
+    * The links are ordered consecutively in anticlockwise order meaning the
+      opposite link is `(link+3)%6`.
+    """
+
+    east = 0
+    north_east = 1
+    north = 2
+    west = 3
+    south_west = 4
+    south = 5
+
+    @classmethod
+    def from_vector(cls, vector):
+        """Given a vector from one node to a neighbour, get the link direction.
+
+        Note that any vector whose magnitude in any given dimension is greater
+        than 1 will be assumed to use a machine's wrap-around links.
+
+        Note that this method assumes a system larger than 2x2 (if a smaller
+        system is provided, the function may produce unexpected or imballenced
+        results).
+
+        Argument
+        --------
+        vector : (x, y)
```

Comment:
The argument is called `vector` (because it is semantically, though not numpily, a vector). I was hoping `(x, y)` sufficiently implies that the type is a tuple? Is there a a method of specifying this which would be clearer?